packages/shared/src/components/MainLayout.tsx

```diff
@@ -158,7 +158,13 @@ export default function MainLayout({
   useEffect(() => {
     if (!shouldRedirectOnboarding) return;
 
-    router.push(`${webappUrl}/onboarding`);
+    const url = new URLSearchParams();
+
+    Object.entries(router.query).forEach(([key, value]) => {
+      url.append(key, value as string);
+    });
+
+    router.push(`${webappUrl}/onboarding?${url.toString()}`);
```

Comment:
Let's only validate that there are query params before we add them. I don't want the url to look weird without a reason :)

Reply:
The browser automatically removes the question mark after the redirection. Though to ensure all browsers, let's do that.